### ISSUE TITLE
[no-use-before-define] Missing allowNamedExports option from base rule

### ISSUE THREAD

**Issue by scottohara** [typescript-eslint/typescript-eslint#5324]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Issue Description

In `eslint@8.17.0`, the `no-use-before-define` rule [learned a new `allowNamedExports` option](https://github.com/eslint/eslint/commit/f6d79209821241c8e03c183b5844a024da0efe8a).

The [docs for the extension rule](https://typescript-eslint.io/rules/no-use-before-define) indicates additional options that extend the base rule options.

However when setting this option on the extention rule, as follows:

```json
"@typescript-eslint/no-use-before-define": ["error", { "allowNamedExports": false }]
```

...the following runtime error is thrown:

```sh
Oops! Something went wrong! :(

ESLint: 8.19.0

Error: .eslintrc.json:
        Configuration for rule "@typescript-eslint/no-use-before-define" is invalid:
        Value {"allowNamedExports":false} should be equal to one of the allowed values.
        Value {"allowNamedExports":false} should NOT have additional properties.
        Value {"allowNamedExports":false} should match exactly one schema in oneOf.
```

On inspection, it appears that the extension rule [defines all options itself](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/rules/no-use-before-define.ts#L245), rather than other extension rules that typically use the `getESLintCoreRule()` machinery to import & extend the schema of the base rule.

I'm not sure if this was intentional, or an oversight.

### Reproduction Repository Link

https://github.com/scottohara/repro-typescript-eslint-no-use-before-define

### Repro Steps

1) clone the repo
2) `yarn install`
3) `eslint test.ts`

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.30.5` |
| `@typescript-eslint/parser`            | `5.30.5` |
| `TypeScript`                           | `4.7.4` |
| `ESLint`                               | `8.19.0` |
| `node`                                 | `18.4.0` |

**Comment by bradzacher** [comment]
> On inspection, it appears that the extension rule defines all options itself, rather than other extension rules that typically use the getESLintCoreRule() machinery to import & extend the schema of the base rule.
>
> I'm not sure if this was intentional, or an oversight.

Intentional because it's a reimplementation of the base rule, it's not just an augmentation.

If we just extended the schema then you would be able to set the option without it doing anything.

---

This just needs to be implemented in our fork by referencing the implementation commit for the base rule.
